### PR TITLE
Improve readiness probe output

### PR DESCRIPTION
**Pull request by steven-sheehy**
Current /ready endpoints don't print enough detail to help users diagnose why health checks are failing. This PR improves the output of the probes and logs.

- Return 200 instead of 204. Some loadbalancers like GKE require probes to return 200 (See #527)
- Return a payload on success since it is no longer 204 "No content"
- Return a helpful error message on failing probe output

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -195,13 +200,17 @@ func (*Ingester) Watch(*grpc_health_v1.HealthCheckRequest, grpc_health_v1.Health
...
 	if err := i.lifecycler.CheckReady(r.Context()); err == nil {
-		w.WriteHeader(http.StatusNoContent)
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write(readinessProbeSuccess)
+		if err != nil {
+			level.Error(util.Logger).Log("msg", "error writing success message", "error", err)
+		}
 	} else {
-		w.WriteHeader(http.StatusInternalServerError)
+		http.Error(w, "Not ready: "+err.Error(), http.StatusInternalServerError)
```
Comment (tomwilkie) [inline pkg/ingester/ingester.go]: Can we reduce the if-nesting here pls?  Something like:

```golang
if err := i.lifecycler.CheckReady(r.Context()); err != nil {
  http.Error(w, "Not ready: "+err.Error(), http.StatusInternalServerError)
  return
}

w.WriteHeader(http.StatusOK)
_, err := w.Write(readinessProbeSuccess)
if err != nil {
  level.Error(util.Logger).Log("msg", "error writing success message", "error", err)
}
```

File: pkg/promtail/server/server.go [hunk 2]
```diff
@@ -156,9 +163,13 @@ func (s *Server) targets(rw http.ResponseWriter, _ *http.Request) {
...
 	if s.tms.Ready() {
-		rw.WriteHeader(http.StatusNoContent)
+		rw.WriteHeader(http.StatusOK)
+		_, err := rw.Write(readinessProbeSuccess)
+		if err != nil {
+			level.Error(logutil.Logger).Log("msg", "error writing success message", "error", err)
+		}
 	} else {
-		rw.WriteHeader(http.StatusInternalServerError)
+		http.Error(rw, readinessProbeFailure, http.StatusInternalServerError)
```
Comment (tomwilkie) [inline pkg/promtail/server/server.go]: Ditto above - reduce nesting please.    See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 for discussion.

File: pkg/ingester/ingester.go [hunk 3]
```diff
@@ -18,6 +19,10 @@ import (
 	"github.com/grafana/loki/pkg/logproto"
 )
 
+var (
+	readinessProbeSuccess = []byte("Ready")
+)
```
Comment (tomwilkie) [inline pkg/ingester/ingester.go]: `var` block not necessary with only one var - prefer:

```golang
var readinessProbeSuccess = []byte("Ready")
```

**Comment by tomwilkie** [comment]
Thanks @steven-sheehy!  Couple of very minor nit picks.

**Comment by steven-sheehy** [comment]
@tomwilkie Thanks for the feedback. I've addressed all the review items. Do you mind reviewing again?

**Pull request merged**